File: LayoutTests/http/tests/cache/resources/prefetched-main-resource-iframe.php
<?php
if ($_SERVER["HTTP_PURPOSE"] == "prefetch") {
    header('Cache-Control: max-age=3600');
    header("Access-Control-Allow-Origin: http://127.0.0.1:8000");

    echo "<script>";
    echo "parent.window.postMessage('FAIL', '*');";
    echo "</script>";

    exit();
}

header("Access-Control-Allow-Origin: http://127.0.0.1:8000");

echo "<script>";
echo "parent.window.postMessage('PASS', '*');";
echo "</script>";
?>

File: LayoutTests/http/tests/cache/resources/prefetched-main-resource-iframe.py
#!/usr/bin/env python3

import os
import sys

purpose = os.environ.get('HTTP_PURPOSE', '')

sys.stdout.write('Content-Type: text/html\r\n\r\n')

if purpose == 'prefetch':
    sys.stdout.write(
        '<script>\n'
        'parent.window.postMessage(\'FAIL\', \'*\');\n'
        '</script>\n'
    )
    sys.exit(0)

sys.stdout.write(
    '<script>\n'
    'parent.window.postMessage(\'PASS\', \'*\');\n'
    '</script>\n'
)